pkg/cmd/create/create.go

```diff
@@ -859,3 +878,137 @@ func defaultAttachFunc(o *CreateOptions, containerToAttach *corev1.Container, si
 		return o.Attach.Attach(req.URL(), o.Config, o.In, o.Out, o.ErrOut, o.TTY, sizeQueue)
 	}
 }
+
+func (o *CreateOptions) watchJobAndStreamLogs(ctx context.Context, clientGetter util.ClientGetter, jobName string) error {
+	k8sclient, err := clientGetter.K8sClientset()
+	if err != nil {
+		return err
+	}
+
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+	defer signal.Stop(sigChan)
+
+	go func() {
+		err = o.watchJobPods(ctx, clientGetter, jobName)
+		if err != nil {
+			fmt.Fprintf(o.Out, "Error watching job pods: %v\n", err)
+		}
+	}()
+
+	jobCompletionErr := make(chan error)
+	go func() {
+		err := o.checkJobCompletion(ctx, k8sclient, jobName)
+		jobCompletionErr <- err
+	}()
+
+	select {
+	case sig := <-sigChan:
+		fmt.Fprintf(o.Out, "Received signal: %v, stopping the job and cleaning up...\n", sig)
+
+		err := k8sclient.BatchV1().Jobs(o.Namespace).Delete(ctx, jobName, metav1.DeleteOptions{
+			PropagationPolicy: ptr.To(metav1.DeletePropagationForeground),
+		})
+		if err != nil {
+			return err
+		}
+
+		return nil
+	case err := <-jobCompletionErr:
+		if err != nil {
+			fmt.Fprintf(o.Out, "Error checking job completion: %v\n", err)
+			return err
+		}
+		fmt.Fprint(o.Out, "Job logs streaming finished.")
+		return nil
+	}
+}
+
+func (o *CreateOptions) watchJobPods(ctx context.Context, clientGetter util.ClientGetter, jobName string) error {
+	k8sclient, err := clientGetter.K8sClientset()
+	if err != nil {
+		return err
+	}
+
+	listOptions := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("job-name=%s", jobName),
+	}
+
+	watcher, err := k8sclient.CoreV1().Pods(o.Namespace).Watch(ctx, listOptions)
+	if err != nil {
+		return err
+	}
+	defer watcher.Stop()
+
+	for event := range watcher.ResultChan() {
+		pod, ok := event.Object.(*corev1.Pod)
+		if !ok {
+			continue
+		}
+
+		if event.Type == watch.Added || event.Type == watch.Modified {
+			if pod.Status.Phase == corev1.PodRunning {
+				go func() {
+					err = o.streamLogsFromPod(ctx, clientGetter, pod.Name)
+					if err != nil {
+						fmt.Fprintf(o.Out, "Error streaming logs from pod %q: %v\n", pod.Name, err)
+					}
+				}()
+			}
+		}
+	}
+	return nil
+}
+
+func (o *CreateOptions) streamLogsFromPod(ctx context.Context, clientGetter util.ClientGetter, podName string) error {
+	k8sclient, err := clientGetter.K8sClientset()
+	if err != nil {
+		return err
+	}
+
+	_, loaded := o.activeStreams.LoadOrStore(podName, true)
+	if loaded {
+		return err
+	}
```

Comment:
looks like a bug, you're returning error from the previous call, maybe this should have it's own `fmt.Errorf` instead of returning the previous error?